interpreter/cling/lib/Interpreter/LookupHelper.cpp

```diff
@@ -84,14 +86,22 @@ namespace cling {
     //
     //  Create a fake file to parse the type name.
     //
-    std::unique_ptr<llvm::MemoryBuffer>
-      SB = llvm::MemoryBuffer::getMemBufferCopy(code.str() + "\n",
-                                                bufferName.str());
+    FileID FID;
+    llvm::hash_code hashedCode = llvm::hash_value(code);
+    auto cacheItr = codeCache.find(hashedCode);
```

Comment:
This cache needs to be cleared (and/or cleaned-up) upon transaction unloading (otherwise it might claim the code is in the AST while it is no longer there ...).

Reply:
IIUC, a FileID and an allocated memory created here are never unloaded and stays in AST. That's why this was biting our pcm size, but maybe I'm wrong. I think @vgvassilev can tell us

Reply:
Maybe we can reuse the FileID but the essence of the unloading is to remove (or probably at this time 'hide') the information that was related brought in by the transaction and thus if the transaction corresponding to a FileID has been unloaded, it must be parsed again if the same code is introduced after the unload.

Reply:
I think I see your point. Currently, that’s not a concern due to the state of the unloading implementation. 

I agree we should assume in this PR rocksolid implementation of unloading. We can probably invalidate our cache of somethig was unloaded. The cons would be that invalidating everything on every event of unloading can be too much and break the performance. Would a good FIXME note address that comment?

Reply:
I don't think so.  Not invalidating the cache upon unloading means that for all intent and purposes we are removing/disabling/breaking-even-more the unloading feature ... for the sake of 'performance'.   I would prefer we spend the extra time now to get this right (or at least to understand the cost of the 'right' solution) rather than making the job of 'fixing' unloading even harder/longer later.

This is even more so true that we are consider multiple caches and re-finding all the cache that need cleanups will become harder and harder.

> I agree we should assume in this PR rocksolid implementation of unloading. 

Please :)

> we can probably invalidate our cache when something was unloaded. T

This is the 'easiest' implementation.  Ideally, we would associate the entries with their transaction and upon unloading the transaction it could 'search' for any part to be removed.
Or we could tie the cache-unloading-reset to the idea of the linear-unloading and simply associate a transaction id to the entries and remove anything with a higher id than the oldest transaction being unloaded.